tests/mocha/field_colour_test.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+suite ('Colour Fields', function() {
+  function assertValue(colourField, expectedValue, expectedText) {
+    var actualValue = colourField.getValue();
+    var actualText = colourField.getText();
+    assertEquals(actualValue, expectedValue);
+    assertEquals(actualText, expectedText);
+  }
+  function assertValueDefault(colourField) {
+    var expectedValue = Blockly.FieldColour.COLOURS[0];
```

Comment:
For testing consistency you might want to change Blockly.FieldColour.COLOURS to a specific set of colours in setup and change it back in teardown.